src/partitioner/local/RDFParser.cpp

```diff
@@ -0,0 +1,422 @@
+
+
+                                DOMNode *child = currentElement2->getFirstElementChild();
+                                DOMElement *childElement
+                                        = dynamic_cast< xercesc::DOMElement * >( child );
+
+
+                                DOMNodeList *dateElements = childElement->getChildNodes();
+                                const XMLSize_t elementCount = dateElements->getLength();
+
+
+                                for (XMLSize_t e = 0; e < elementCount; ++e) {
+
+                                    DOMNode *dateEle = dateElements->item(e);
+                                    if (dateEle->getNodeType() &&
+                                        dateEle->getNodeType() == DOMNode::ELEMENT_NODE) // is element
+                                    {
+
+
+                                        DOMElement *currentDateElement = dynamic_cast< xercesc::DOMElement * >( dateEle );
+                                        if (XMLString::equals(currentDateElement->getTagName(), TAG_year_of)) {
+
+                                            const XMLCh *yearOf = currentDateElement->getTextContent();
+                                            year = XMLString::transcode(yearOf);
+                                            attributes[5] = year;
+
+                                        } else if (XMLString::equals(currentDateElement->getTagName(), TAG_month_of)) {
+
+                                            const XMLCh *monthOf = currentDateElement->getTextContent();
+                                            month = XMLString::transcode(monthOf);
+                                            attributes[4] = month;
+
+                                        }
+                                    }
+                                }
+
+
+                            } else if (XMLString::equals(currentElement2->getTagName(), TAG_has_web_address)) {
+
+                                const XMLCh *web_add = currentElement2->getTextContent();
+                                web_address = XMLString::transcode(web_add);
+                                attributes[3] = web_address;
+
+
+                            }
+
+                        }
+                    }
+
+
+                }
+
+                if (attributes->size() > 0) {
+                    articlesMap.insert({articleID, attributes});
+
+                }
+                for (int i = 0; i < authorsInArticle.size(); i++) {
+
+                    for (int j = i + 1; j < authorsInArticle.size(); j++) {
+
+                        edgelist.push_back(make_pair(authorsInArticle.at(i), authorsInArticle.at(j)));
+                        addToEdges(&edgeMap, authorsInArticle.at(i), authorsInArticle.at(j), articleID);
+
+                        file << authorsInArticle.at(i) << ' ' << authorsInArticle.at(j);
+
+                    }
+                }
+
+                printEdges();
```

Comment:
This is debugging statement isn't it? We should remove this.

Reply:
No. It is for printing edge list to a file. It is the file reading by metis.

Reply:
Ok, then lets change "printEdges()" to "writeEdgesToFile()"